utils/mil_tools/nodes/label_me_bag.py

```diff
@@ -0,0 +1,231 @@
+from mil_msgs.msg import LabeledObjects, LabeledObject
+
+class BagToLabelMe():
+  """
+  Interfaces between ROS bags and LabelMe images.
+  
+  Can be used to insert images from ROS bags into LabelMe instalation
+  or extract LabelMe annotations into a ROS bag
+  
+  TODO: improve verbosity, disable print statements with toggle
+  """
+  def __init__(self, config, labelme_dir):
+      """
+      Generates class that can be used to insert or extract images to/from LabelMe
+      
+      config: path to YAML file following format of the example yaml file in this package
+      labelme_dir: root directory of labelme instalation (should contain Images and Annotations directories)
+      """
+      if not os.path.isdir(labelme_dir):
+          raise Exception("Labelme directory {} does not exsist".format(labelme_dir))
+
+      config_file = open(config, 'r')
+      self.config = yaml.load(config_file)
+      self._verify_yaml()
+      self.bridge = CvBridge()
+      self.labelme_dir = labelme_dir
+
+  def read_bags(self):
+      """
+      Crawls through all bags in config YAML, placing images from the specified
+      topics and the specified frequency into corosponding folders within
+      Images of the labelme instalation.
+      
+      These folders will be created as follows if not already:
+      LABELME_DIR/Images/SEGEMENT_NAME/TOPIC_NAME
+      where TOPIC_NAME replaces all / with @ for compatibility with filesystems.
+      """
+      for bag in self.config['bags']:
+          print "Opening {}".format(bag['file'])
+          self._read_bag(bag['file'], bag['segments'])
+
+  def extract_labels(self):
+      """
+      Finds labeled XML files in LabelMe directories determined from the
+      config file, placing parsed annotation data as ros messages in a new bag
+      """
+      print "Extracting labels for all bags in config"
+      for bag in self.config['bags']:
+          print "\tExtracting LabelMe annotations for {}".format(bag['file'])
+          # TODO: give output bag a default name like INFILE_labeled.bag
+          outfile = bag['labeled_file']
+          self._extract_labels_bag(bag, outfile)
+
+  def print_report(self):
+      """
+      Compares how many images are present in each LabelMe directory affected
+      by the config YAML to the number of annotated XML files produced by
+      LabelMe in each of these directories.
+      
+      Prints out these counts in percentages by segement, bag, and overall
+      """
+      print "Generating completion report for all bags in config"
+      total_xml_count = 0
+      total_img_count = 0
+      for bag in self.config['bags']:
+          print "\tGenerating completion report for {}".format(bag['file'])
+          x, i = self._print_bag_report(bag)
+          total_xml_count += x
+          total_img_count += i
+      print "{}/{} TOTAL images labeled ({}%)".format(total_xml_count, total_img_count, round(100.0*total_xml_count/total_img_count,2))
+
+  def _verify_yaml(self):
```

Comment:
Maybe have this function print the intent of the script, i.e. "I will dump images from {topic} to {outdir} on {bagname} from {start} to {end} every {freq} seconds." for each entry in the yaml. I know you do this down in the bag function, but having it all print out at once may be more useful. That would let you verify everything will run smoothly and leave your computer and let it work (I don't know how long this takes to run but on multiple big bags it could be rough)

This would have been cool for the mission runner back in the good old days of Navigator.